Gordon360/ApiControllers/JobsController.cs

```diff
@@ -0,0 +1,223 @@
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e.Message);
+                return Request.CreateResponse(HttpStatusCode.InternalServerError, e);
+            }
+            return Request.CreateResponse(HttpStatusCode.OK, result);
+        }
+
+        /// <summary>
+        /// Get a user's active jobs
+        /// </summary>
+        /// <returns>The result of deleting the shift</returns>
+        [HttpDelete]
+        [Route("deleteShift/{rowID}")]
+        [StateYourBusiness(operation = Operation.DELETE, resource = Resource.SHIFT)]
+        public IHttpActionResult deleteShiftForUser(int rowID)
+        {
+            IEnumerable<StudentTimesheetsViewModel> result = null;
+            int userID = -1;
+
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+            userID = Convert.ToInt32(id);
+
+            try
+            {
+                result = _jobsService.deleteShiftForUser(rowID, userID);
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e.Message);
+                return InternalServerError();
+            }
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Submit shifts
+        /// </summary>
+        /// <returns>The result of deleting the shift</returns>
+        [HttpPost]
+        [Route("submitShifts")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.SHIFT)]
+        public IHttpActionResult submitShiftsForUser([FromBody] IEnumerable<ShiftToSubmitViewModel> shifts)
+        {
+            IEnumerable<StudentTimesheetsViewModel> result = null;
+
+            try
+            {
+                foreach (ShiftToSubmitViewModel shift in shifts)
+                {
+                    System.Diagnostics.Debug.WriteLine("Submitting shift to " + shift.SUBMITTED_TO);
+                    result = _jobsService.submitShiftForUser(shift.ID_NUM, shift.EML, shift.SHIFT_END_DATETIME, shift.SUBMITTED_TO, shift.LAST_CHANGED_BY);
+                }
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e.Message);
+                return InternalServerError();
+            }
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Submit shifts
+        /// </summary>
+        /// <returns>The result of deleting the shift</returns>
+        [HttpGet]
+        [Route("supervisorName/{supervisorID}")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.SHIFT)]
+        public IHttpActionResult submitShiftsForUser(int supervisorID)
+        {
```

Comment:
Should this and the above both have `public IHttpActionResult submitShiftsForUser`? This isn't the function name is it?

Reply:
Yikes! I didn't see that one at all. Uh, that's a stupid mistake on my part... I'll get on that.